src/store/MenuStore.js

```diff
@@ -30,10 +30,19 @@ export default class MenuStore {
         stx.allowScroll = stx.allowZoom = !this.open;
     }
 
+
     @action.bound onTitleClick(e) {
+        const enableRouting = this.mainStore.chart.enableRouting;
         if (e) {
             e.stopPropagation();
         }
+        if (enableRouting && !this.open && this.route) {
+            window.history.replaceState({ urlPath:'#' }, '', '#');
+            window.history.pushState({ urlPath:`#${this.route}` }, '', `#${this.route}`);
+        } else if (enableRouting && this.open && this.route) {
+            window.history.replaceState({ urlPath:'#' }, '', '#');
+        }
```

Comment:
@shayanbinary move this logic to RoutingStore

Reply:
@aminroosta this happen when a user click to open a dialog, you mean on `menuStore` I should call a `RoutingStore` method to handle this ?